graphql-ballerina/engine.bal

```diff
@@ -27,22 +27,23 @@ isolated class Engine {
         self.maxQueryDepth = getMaxQueryDepth(serviceConfig);
     }
 
-    isolated function getOutputObjectForQuery(string documentString, string operationName) returns OutputObject {
+    isolated function getOutputObjectForQuery(string documentString, string operationName) returns
+                                                                                        [OutputObject, string] {
         parser:DocumentNode|OutputObject result = self.parse(documentString);
         if (result is OutputObject) {
-            return result;
+            return [result, "SyntaxError"];
```

Comment:
It is better to define constants for the error codes and use them here instead of using strings directly. Otherwise, if we have to change the error code, we have to change every occurrence of this string in the code.